lib/generators/json_serializable_generator.dart

```diff
@@ -223,14 +229,58 @@ class JsonSerializableGenerator
         // ...resetting `isList` to `false`.
         isList = false;
       }
-    }
 
-    if (!isList && _coreIterableChecker.isAssignableFromType(fieldType)) {
-      // Then we need to add `?.toList()
-      expression += "?.toList()";
+      if (!isList) {
+        // Then we need to add `?.toList()
+        expression += "?.toList()";
+      }
+
+      return expression;
+    } else if (_coreMapChecker.isAssignableFromType(fieldType)) {
+      var args = _getTypeArguments(fieldType, _coreMapChecker);
+      assert(args.length == 2);
+
+      var keyArg = args.first;
+      var valueType = args.last;
+
+      // We're not going to handle converting key types at the moment
+      // So the only safe types for key are dynamic/Object/String
+      var safeKey = keyArg.isDynamic ||
+          keyArg.isObject ||
+          _coreStringChecker.isExactlyType(keyArg);
+
+      var safeValue = false;
+      if (valueType.isDynamic || valueType.isObject) {
+        safeValue = true;
+      } else {
+        safeValue = _isStringBoolNum(valueType);
+      }
+
+      if (safeKey) {
```

Comment:
there is no else to match here, should it be throwing or is it intended to fall through to the unsafe fallback case?